pkg/network/tcp_peer.go

```diff
@@ -103,6 +130,27 @@ func (p *TCPPeer) handleConn() {
 	p.Disconnect(err)
 }
 
+// handleQueues is a goroutine that is started automatically to handle
+// send queues.
+func (p *TCPPeer) handleQueues() {
+	var err error
+
+	for {
+		var msg []byte
+		select {
+		case <-p.done:
+			return
+		case msg = <-p.hpSendQ: // High priority first.
```

Comment:
I am not sure if select guarantees that this receive will happen before the next in case both are available.